linera-explorer/src/input_type.rs

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) Zefchain Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use super::js_utils::{getf, unproxy};
+use wasm_bindgen::prelude::*;
+
+/// add a input line for lists
+#[wasm_bindgen]
+pub fn append_input(component: JsValue) {
+    let t = getf(&component, "t");
+    let input: js_sys::Array = getf(&t, "_input").into();
+    let child = unproxy(&getf(&t, "ofType"));
+    input.splice(input.length(), 0, &child);
```

Comment:
Would `input.push(&child)` make more sense here? (Or is that not equivalent?)

Reply:
I guess the internal changes are different and splice is required for Vue to update the data for an array